### PR TITLE
BDRSPS-966 SSD Site and SiteVisit IRI construction

### PR DESCRIPTION
**Pull request by Lincoln-GR**
Add pattern functions for constructing Site and Site Visit IRIs.
This should be the last changes required in the survey occurrence template

### CODE REVIEW

File: abis_mapping/templates/survey_occurrence_data_v2/mapping.py [hunk 1]
```diff
@@ -647,15 +638,25 @@ def apply_mapping_row(
...
 
+        # Create URIs for Survey-related fields (i.e. fields not on the incidental template)
+
         # Create TERN survey IRI from surveyID field
         survey_id: str | None = row["surveyID"]
         survey = utils.iri_patterns.survey_iri(base_iri, survey_id)
 
+        # Create Tern Site IRI, depending on the siteID field
+        site_id: str | None = row["siteID"]
+        if site_id:
```
Comment (joecrowleygaia) [inline abis_mapping/templates/survey_occurrence_data_v2/mapping.py:649]: not a walrus enjoyer?

**Review by joecrowleygaia — APPROVED** [review]

File: abis_mapping/templates/survey_occurrence_data_v2/mapping.py (hunk 1 above)
Comment (Lincoln-GR) [reply abis_mapping/templates/survey_occurrence_data_v2/mapping.py:649]: sometimes yes, but here the walrus was preventing me from adding a proper type annotation to the variables like `site_id`

Comment (joecrowleygaia) [reply abis_mapping/templates/survey_occurrence_data_v2/mapping.py:649]: i probs didn't notice its use somewhere mypy complained.

Comment (Lincoln-GR) [reply abis_mapping/templates/survey_occurrence_data_v2/mapping.py:649]: Well without the type annotation `row["field"]` is `Any`, and `Any` means "don't type check this", so mypy generally won't complain about that

**Pull request merged**